include/bout/utils.hxx

```diff
@@ -362,6 +362,14 @@ public:
     return data[i.ind];
   }
 
+  T& operator[](Ind3D i) {
+    // ny and nz are private :-(
+    // ASSERT2(i.nz == n3);
+    // ASSERT2(i.ny == n2);
+    ASSERT2(0 <= i.ind && i.ind < n1 * n2 * n3);
```

Comment:
I do not understand how that should be simplified. Is that a false positive?

Reply:
It's because it's seeing through the macro and trying to simplify the `(!expression)`, but it should be ignored by `readability-simplify-boolean-expr.IgnoreMacros = true` in `.clang-tidy, so I'm not sure what's going on here